shuup/admin/modules/orders/views/list.py

```diff
@@ -91,3 +91,161 @@ def get_object_abstract(self, instance, item):
+
+    def format_taxful_total_price(self, instance, *args, **kwargs):
+        return escape(format_money(instance.taxful_total_price))
+
+    def label(self, instance, *args, **kwargs):
+        # format label to make it human readable
+        return instance.label.replace("_", " ").title()
+
+    def get_object_abstract(self, instance, item):
+        return [
+            {"text": "%s" % instance, "class": "header"},
+            {"title": _(u"Total"), "text": item.get("taxful_total_price_value")},
+            {"title": _(u"Status"), "text": item.get("status")}
+        ]
+
+
+class ShipmentListView(PicotableListView):
+    model = Shipment
+    default_columns = [
+        Column("id", _(u"ID"), linked=True),
+        Column("order_id", _(u"Order"), display="get_order_name", raw=True),
+        Column("get_content", _(u"Content"), display="get_content", raw=True),
+        Column("supplier_id", _(u"Supplier"), display="get_supplier", raw=True),
+        Column("tracking_code", _(u"Tracking Code"), display="tracking_code_url", raw=True),
+        Column("status", _(u"Status"), display="create_action_buttons", raw=True),
+    ]
+
+    def get_context_data(self):
+        context = super(ShipmentListView, self).get_context_data()
+        context["listview"] = True
+        return context
+
+    def tracking_code_url(self, instance):
+        if instance.tracking_url:
+            return f"<a href='{instance.tracking_url}'>{instance.tracking_code}</a>"
+        return instance.tracking_code
+
+    def get_supplier(self, instance):
+        return instance.supplier.name
+
+    def get_order_name(self, instance):
+        return instance.order.__str__()
+
+    def get_content(self, instance):
+        query = ShipmentProduct.objects.filter(shipment=instance)
+        product = ""
+        for i in query:
+            product = product + f"{i.product} ({i.quantity})"
+        return product
+
+    def create_action_buttons(self, instance):
+        if instance.status == ShipmentStatus.SENT:
+            return instance.status.name
+        return f'<a class="btn btn-info btn-sm set_shipment_status" \
+            href="{reverse("shuup_admin:order.shipments-list")}" shipment_id="{instance.id}" \
+            onclick="handleSetSent(event, this)"><i class="fa fa-truck"></i> Mark as sent</a>'
+
+    def __init__(self):
+        super(ShipmentListView, self).__init__()
+        self.columns = self.default_columns
+    def get_queryset(self):
+        if get_supplier(self.request) is None:
+            return super(ShipmentListView, self).get_queryset().exclude(status=ShipmentStatus.DELETED)
+        return super(ShipmentListView, self).get_queryset().exclude(status=ShipmentStatus.DELETED).filter(supplier=get_supplier(self.request))
+
+    def get_object_abstract(self, instance, item):
+        return [
+            {"text": "%s" % instance, "class": "header"},
+            {"title": _(u"Content"), "text": item.get("get_content")},
```

Comment:
make sure to also show the status, the button to set as sent and other columns as well